src/goose.rs

```diff
@@ -488,9 +488,11 @@ impl GooseTaskSet {
     /// ```
     pub fn set_weight(mut self, weight: usize) -> Result<Self, GooseError> {
         trace!("{} set_weight: {}", self.name, weight);
-        if weight < 1 {
-            error!("{} weight of {} not allowed", self.name, weight);
-            return Err(GooseError::InvalidWeight);
+        if weight == 0 {
```

Comment:
Oh - usize ensures it won't ever be unsigned -- good catch.

Reply:
Clippy has caught this for me in the past, I don't know why it didn't complain about this one.